### DISCUSSION TITLE
"Not a variable name" after changing the Start g-code

### DISCUSSION THREAD

**Discussion by bjoernbarth** [supermerill/SuperSlicer#3847] [Q&A]
Hello,
i want to change the start g-code but it doesn't work. How can I solve the problems? I use the g-code flavor "Klipper".

Start g-code:

# Get Params
  {% set t_extruder = params.T_EXTRUDER|default(0)|float %}
  {% set t_bed = params.T_BED|default(0)|float %}

  # Start
  M104 S{t_extruder}                 ; Setze Hotend Temperatur
  M140 S{t_bed}                      ; Setze Heatbed Temperatur
  G28                                ; Home
  M83                                ; Extruder relativer Modus
  G92 E0.0                           ; Extruder Reset
  G90                                ; Absolute Positionierung
  M190 S{t_bed}                      ; Warte auf Bed Zieltemperatur
  M109 S{t_extruder}                 ; Warte auf Hotend Zieltemperatur
  G1   Z0

![IMG_0134](https://github.com/supermerill/SuperSlicer/assets/61991527/5d5a8b1a-97da-4c29-bf37-6f36204be34c)
![IMG_0133](https://github.com/supermerill/SuperSlicer/assets/61991527/457974bb-a704-47a7-98b1-49d17e6ebd27)

**Comment by theophile** [comment]
That's not g-code, it's jinja2 syntax for klipper macros. The slicer has no idea what to do with that. You need to write the klipper macro then put the command for that macro in the custom g-code field.
